runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -1654,17 +1663,26 @@ class INTERPRETER_CLASS
 
 		if (J9_OBJECT_MONITOR_ENTER_FAILED(monitorRC)) {
 			*bp |= J9SF_A0_INVISIBLE_TAG;
+			switch (monitorRC) {
 #if JAVA_SPEC_VERSION >= 16
-			if (J9_OBJECT_MONITOR_VALUE_TYPE_IMSE == monitorRC) {
+			case J9_OBJECT_MONITOR_VALUE_TYPE_IMSE:
 				_currentThread->tempSlot = (UDATA) syncObject;
 				rc = THROW_VALUE_TYPE_ILLEGAL_MONITOR_STATE;
-			} else
+				break;
 #endif /* JAVA_SPEC_VERSION >= 16 */
-			{
+#if defined(J9VM_OPT_CRIU_SUPPORT)
+			case J9_OBJECT_MONITOR_CRIU_SINGLE_THREAD_MODE_THROW:
+				rc = THROW_CRIU_SINGLE_THREAD_MODE;
+				break;
+#endif /* defined(J9VM_OPT_CRIU_SUPPORT) */
+			case J9_OBJECT_MONITOR_OOM:
 				/* Monitor was not entered - hide the frame to prevent exception throw from processing it.
 				 * Note that BP can not have changed during a failed enter.
 				 */
 				rc = THROW_MONITOR_ALLOC_FAIL;
+				break;
+			default:
```

Comment:
we dont need this unreachable block

Reply:
Last code standard doc I had says _Switch statements should always include a default: label, if only to assert_.

Reply:
ah okay, you can keep it then. I was mostly worried about adding more code to the interpreter. Little things can have a perf impact.